middleware/auth/jwt/jwt.go

```diff
@@ -77,7 +87,6 @@ func WithTokenHeader(header map[string]interface{}) Option {
 func Server(keyFunc jwt.Keyfunc, opts ...Option) middleware.Middleware {
 	o := &options{
 		signingMethod: jwt.SigningMethodHS256,
-		claims:        jwt.RegisteredClaims{},
 	}
 	for _, opt := range opts {
 		opt(o)
```

Comment:
```
claims := func() jwt.Claims {
    return o.claims
}
```
u can convert the claims to func factory

Reply:
o.claims is already a func, how about changing it to the following code?
```go
o := &options{
	signingMethod: jwt.SigningMethodHS256,
	claims: func() jwt.Claims {
		return jwt.MapClaims{}
	},
}
```

Reply:
do not add two different options (`WithClaims`/`WithServerClaims`), u should only use `WithClaims` option.

Reply:
but `WithServerClaims` need a factory func, `WithClaims` just need a single jwt.Claims object. If combined into one method, you must modify the api of `WithClaims` like `func WithClaims(f func() jwt.Claims) Option`

Reply:
modify the api of WithClaims to func WithClaims(f func() jwt.Claims) option, it looks ok.

Reply:
Thanks, then I can change it with confidence.